internal/plugincommon/ccipreader.go

```diff
@@ -0,0 +1,121 @@
+package plugincommon
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-ccip/internal/reader"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+)
+
+type BackgroundReaderSyncer struct {
+	lggr          logger.Logger
+	reader        reader.CCIP
+	syncTimeout   time.Duration
+	syncFrequency time.Duration
+
+	bgSyncCtx context.Context
+	bgSyncCf  context.CancelFunc
+	bgSyncWG  *sync.WaitGroup
+}
+
+func NewBackgroundReaderSyncer(
+	lggr logger.Logger,
+	reader reader.CCIP,
+	syncTimeout time.Duration,
+	syncFrequency time.Duration,
+) *BackgroundReaderSyncer {
+
+	return &BackgroundReaderSyncer{
+		lggr:          lggr,
+		reader:        reader,
+		syncTimeout:   syncTimeout,
+		syncFrequency: syncFrequency,
```

Comment:
nit/ Do we need to add some validation here? (i.e. syncTimeout > syncFrequency)?

Same for loggr and reader not nil?